### PR TITLE
Integration test for condition choices

### PR DESCRIPTION
**Pull request by Hurleveur**
As said in the title, hopefully this works

~~DO NOT MERGE THIS UNTIL THE TESTS RUN GREEN~~ all is good they are green and I tested it offline too

### CODE REVIEW

**Comment by Hurleveur** [comment]
I merged master because otherwise it wasnt going to run the integration tests as that was just fixed
edit: I wish this would just run the tests automatically...

**Comment by SolraBizna** [comment]
I don't know if it's been helping, but I've been clicking the button to manually allow the tests to run here. I can't keep doing that because I'm going to bed. >_>

**Comment by Hurleveur** [comment]
It seems like it does not work because of the parsing being weird and not understanding the "to show"... I can't do much about that?

edit: Its to display not to show, plz update the PR description.

**Comment by Hurleveur** [comment]
I tested it manually I can now affirm this works.

File: tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+
+mission "%TEST%: CONDITIONAL CHOICE"
+	priority
+	source "Ruin"
+	to offer
+		has "%TEST%: CONDITIONAL CHOICE"
+	on offer
+		conversation
+			# cant actually write an integration test for seeing this or not
```
Comment (petervdmeer) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt:65]: Add one with a `goto` that goes to some action that sets a variable. (Then you can check if the `goto` happened.)

File: tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+				`	"No."`
+					to display
+						has "%TEST%: CONDITIONAL CHOICE"
+					accept
+				`	"Yes."`
+					decline
+	on accept
+		set "succeed test"
+
+test "Land On Ruin helper"
```
Comment (petervdmeer) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt]: This helper function is already in master (created by you in the other PR), no need to repeat it.

Comment (Hurleveur) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt]: https://github.com/endless-sky/endless-sky/pull/7623 didnt get merged yet?

Comment (Hurleveur) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt]: When it will be it should be removed for sure though

**Comment by Hurleveur** [comment]
Here we go, should be good to be merged now (this succeeded the tests as a quick PR on the main branch already but feel free to run them again if you want)

**Comment by SolraBizna** [comment]
I'll want to take a closer look before merging, but I expect to get that done within 24 hours. (If I don't have time, or if it gets another passing review, I'll just hit merge.)

**Comment by UnorderedSigh** [comment]
Do you have a "set the player's name" test in there? The conversation code uses a choice with no options to represent "set the player's name." The code has to differentiate between a choice with no options, and a choice that has options, all of which are hidden. That works in my version, that was merged in #3.

**Comment by UnorderedSigh** [comment]
I'd feel preposterous doing a PR to a PR to a PR, so I'll hope one of you adds a "name" test.

**Comment by Hurleveur** [comment]
I did a new test for that, and I'll even open one of the like to master. Do note I have not tested it yet so I recommend running the tests before merging.

**Comment by SolraBizna** [comment]
> I'd feel preposterous doing a PR to a PR to a PR, so I'll hope one of you adds a "name" test.

"Two levels of PR, shame on you. Three or more levels of PR, shame on me." - Mark Twain, probably

> Do note I have not tested it yet so I recommend running the tests before merging.

I hit the button. At some point, I should figure out how to make it so that you don't need me to hit the button. Or, more generally, how any of this CI stuff works. Or if me hitting the button even has any use.

**Comment by Hurleveur** [comment]
Well once I'll have one merged PR on your branch it'll run automatically
if you want just make a dummy branch I'll PR smth of no importance to it, which you would then merge and then the tests will run automatically

**Comment by Hurleveur** [comment]
Nice, the tests run green indeed!
Time to merge I think

**Pull request merged**